tests/system/providers/amazon/aws/example_sagemaker_endpoint.py

```diff
@@ -88,6 +88,11 @@ def delete_endpoint(endpoint_name):
     boto3.client("sagemaker").delete_endpoint(EndpointName=endpoint_name)
 
 
+@task(trigger_rule=TriggerRule.ALL_DONE)
+def archive_logs(log_group_name):
+    boto3.client("logs").put_retention_policy(logGroupName=log_group_name, retentionInDays=1)
```

Comment:
Is 1 day would be enough? In some previous PR which also related to retention period for CloudWatch logs in system tests (can't find it quick) we decide as far a i remember 14 days

More expensive it write Gigabytes/Terabytes of logs rather than store it

Reply:
Good question. My thinking was, we currently remove the log group so let's put the smallest retention period possible (that is one day). Your question brings a project/idea we were thinking working on, which is to keeps logs of resources which have been created in a failed system test. That way, it could simplify investigation upon failure. But I think we should come up with a solution for all system tests. That should be another PR to me

Reply:
Finally found https://github.com/apache/airflow/pull/28819#discussion_r1069787616 previous discussion

Reply:
The only issue here is `_purge_logs` do not work because of the reason explained in the description. Even though the log group is removed, it gets recreated because some logs are still being pushed after the system test ends. The solution I can think of is to add a parameter `successful_test_action` (or a better name) with the default value being `_purge_logs`. That way, for this test, we could pass `archive_logs` as value of `successful_test_action`

Reply:
Oh... got you. That a bit strange Sagemaker behaviour that doesn't allow specify name of Log Group and it create only [predefined](https://docs.aws.amazon.com/sagemaker/latest/dg/logging-cloudwatch.html)?

So in real environment `/aws/sagemaker/Endpoints/[EndpointName]` in the end could hit limit of Log Group per region per account (looong-long time) or keep enormous number of  Log Groups in Cloudwatch 🙄 
Because even if Log Streams would removed by retention period the Log Group still exists in account.